charts/tigera-operator/values.yaml

```diff
@@ -5,18 +5,79 @@
 # Example: --set-file imagePullSecrets.gcr=./pull-secret.json
 imagePullSecrets: {}
 
+# Configures general installation parameters for Calico. Schema is based
+# on the operator.tigera.io/Installation API documented
+# here: https://docs.tigera.io/calico/latest/reference/installation/api#installationspec
 installation:
   enabled: true
-  kubernetesProvider: ""
+
+  # Disable CSI driver. Default: /var/lib/kubelet
+  kubeletVolumePluginPath: "None"
+
+  # --- General Image Settings ---
+  registry: ""
+  imagePath: ""
+  imagePrefix: ""
+
   # imagePullSecrets are configured on all images deployed by the tigera-operator.
   # secrets specified here must exist in the tigera-operator namespace; they won't be created by the operator or helm.
   # imagePullSecrets are a slice of LocalObjectReferences, which is the same format they appear as on deployments.
   #
   # Example: --set installation.imagePullSecrets[0].name=my-existing-secret
   imagePullSecrets: []
 
-  # Disable CSI driver by default.
-  kubeletVolumePluginPath: "None"
+  # Valid options: EKS, GKE, AKS, RKE2, OpenShift, DockerEnterprise, TKG, Kind.
+  # If empty the operator will attempt to automatically determine the current provider.
+  kubernetesProvider: ""
+
+  cni:
```

Comment:
This is another instance where we actually do "magic" defaulting based on the platform, so I think this section should be left commented as well.

For example, the operator code detects the CNI plugin based on the platform provider here: https://github.com/tigera/operator/blob/076feeb163410f4bbb700270f802b6e31c989312/pkg/controller/installation/core_controller.go#L487-L502